### PR TITLE
feat(TD-21): Edit tasks in the backend

### PR DESCRIPTION
**Pull request by taiman-siddiqui-sazim**
**Description of change**

- Tasks can be edited in the backend
- Edited tasks are fetched from the backend

**Associated Trello link**

https://trello.com/c/MoGg6Qme

**Screenshot of change**

I can use the following curl command to edit a task in the backend:

```
curl -X PATCH http://localhost:3001/api/v1/tasks/61 \
-H "Content-Type: application/json" \
-d '{
  "task": {
    "title": "Edit and fetch task"
  }
}'
```

The task is edited and fetched from the backend:

Screenshot 2024-12-19 at 6 32 15 AM

**Dev notes**

- Frontend has been updated to edit and fetch tasks using this API
- Features to delete and mark tasks as completed has not been implemented yet

### CODE REVIEW

File: app/controllers/api/v1/tasks_controller.rb [hunk 1]
```diff
@@ -35,14 +35,43 @@ def create
...
+            ['Only task title can be updated'],
+            metadata: metadata
+          ), status: :unprocessable_entity
+          return
+        end
+      
+        @task.assign_attributes(task_update_params)
+        validator = TaskValidator.new(@task)
+      
+        if validator.valid? && @task.save
```
Comment (ishmam-iqbal-sazim) [inline app/controllers/api/v1/tasks_controller.rb:50]: If you're validating in the Task model itself, you probably don't need the validator here

File: app/controllers/api/v1/tasks_controller.rb [hunk 2]
```diff
@@ -35,14 +35,43 @@ def create
...
       end
 
+      def update
+        if params[:task].keys.any? { |key| !task_update_params.key?(key) }
+          render json: TaskSerializer.serialize_errors(
+            ['Only task title can be updated'],
+            metadata: metadata
+          ), status: :unprocessable_entity
+          return
+        end
```
Comment (ishmam-iqbal-sazim) [inline app/controllers/api/v1/tasks_controller.rb:45]: I don't think this section is necessary as you're already whitelisting title in the update task params method.
Consider removing this

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

File: app/controllers/api/v1/tasks_controller.rb (hunk 2 above)
Comment (taiman-siddiqui-sazim) [reply app/controllers/api/v1/tasks_controller.rb:45]: I include this section to return an error if any non-permitted parameters are even attempted to be submitted

**Pull request merged**